types/audio-context/index.d.ts

```diff
@@ -0,0 +1,30 @@
+// Type definitions for audio-context 1.0
+// Project: https://github.com/audiojs/audio-context
+// Definitions by: Jeff Peterson <https://github.com/bdjeffyp>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/**
+ * Options for your audio context:
+ * @param sampleRate if specified, will set the context sampleRate.
+ * @param latencyHint if specified, will control latency. One of 'balanced', 'playback', 'interaction' (default) or number.
+ * @param offline if specified, will create OfflineAudioContext.
+ * @param length if specified, will set number of frames for offline context.
+ * @param channels if specified, will set number of channels for offline context.
+ * @param contextAttributes any other options for the context.
+ */
+export interface Options {
+    sampleRate?: number;
+    latencyHint?: string | number;
+    offline?: boolean;
+    length?: number;
+    channels?: number;
+    contextAttributes?: object;
+}
+
+/**
+ * Gets an audio context from your web browser.
+ * @param options Takes an Options object or just provide a sample rate.
+ * @returns the audio context or null if there was an error or not a web browser.
+ */
+export function getContext(options?: Options | number): AudioContext | null;
```

Comment:
It looks like, to match the underlying library, you want to drop all `export` keywords and then add `exports = getContext;` at the bottom.

Reply:
I had thought about that, but doing so affects the accessibility of the Options interface. I wouldn't be able to export the interface then because of the `export = getContext'`. I feel that the interface is important for users to have since it defines exactly what the `getContext()` method is expecting to get.
I couldn't find any examples in my searches for a solution to that problem. Do you have any recommendations?

Reply:
Nevermind. I figured out what I was doing wrong after another look at the module-function template. 🤦‍♂️😅